### ISSUE TITLE
items are duplicated

### ISSUE THREAD

**Issue by frogglen** [BenWoodworth/FastCraft#171]
## Description

Well, I tried to make a new recipe with the RecipeManager plugin and I was able to do it correctly resulting in a named item and custom lore. The FastCraft plugin I take it and when obtaining the item from FastCraft, the ingredients do not consume them and you can do the crafting infinitely.

## Server Details
- Server:  Paper 1.15.2 #273
- FastCraft: v3.1.3

I have RecipeManager v2.22.1

**Comment by BenWoodworth** [comment]
I can take a look at this in about an hour. Could you share the recipe with
me (the file it's saved in if possible) so I can reproduce this?

**Comment by frogglen** [comment]
the recipe i made was this:

craft
wheat + wheat + wheat
wheat + poisonous_potato + wheat
nether_wart + nether_wart + nether_wart
= wheat_seeds:0:5
@name Alimento Economico
@lore Rango Bajo
@lore Sirve para alimentar Mascotas Iniciales.

you can replicate it by placing it in the recipemanager recipe folder

**Comment by BenWoodworth** [comment]
Alright, I was able to reproduce it. I'll take a look

Before & after crafting with FastCraft:
![2020-05-10_16 29 31](https://user-images.githubusercontent.com/1829157/81509860-9dbb1c80-92db-11ea-855f-0867c90eef99.png)
![2020-05-10_16 29 34](https://user-images.githubusercontent.com/1829157/81509865-a1e73a00-92db-11ea-99fa-39734c2c852e.png)

**Comment by frogglen** [comment]
> Alright, I was able to reproduce it. I'll take a look
> 
> Before & after crafting with FastCraft:
> ![2020-05-10_16 29 31](https://user-images.githubusercontent.com/1829157/81509860-9dbb1c80-92db-11ea-855f-0867c90eef99.png)
> ![2020-05-10_16 29 34](https://user-images.githubusercontent.com/1829157/81509865-a1e73a00-92db-11ea-99fa-39734c2c852e.png)

exactly, the resources are not spent to make the recipe, but if I do it on the normal preparation table, the custom recipe works and the materials are consumed.

**Comment by BenWoodworth** [comment]
It looks like RecipeManager is doing something weird with the crafting events

RecipeManager is cancelling the crafting (so FastCraft won't take the ingredients, and won't give the seeds), then it's giving the players seeds itself.

I'll try to figure out how to make that work correctly, but for now, I released v3.1.4 which disables the recipes from RecipeManager

**Comment by frogglen** [comment]
> It looks like RecipeManager is doing something weird with the crafting events
> 
> RecipeManager is cancelling the crafting (so FastCraft won't take the ingredients, and won't give the seeds), then it's giving the players seeds itself.
> 
> I'll try to figure out how to make that work correctly, but for now, I released v3.1.4 which disables the recipes from RecipeManager

thanks, i have temporarily disabled fastcraft on my server and will install the version you mention to me

**Comment by BenWoodworth** [comment]
RecipeManager handles the CraftItemEvent:
https://github.com/haveric/RecipeManager2/blob/8b3884a27bee3c268bb4e8a642383bee4be4835a/RecipeManager-base/src/main/java/haveric/recipeManager/recipes/WorkbenchEvents.java#L325

RecipeManager adds the result (seeds) to the player's inventory:
https://github.com/haveric/RecipeManager2/blob/8b3884a27bee3c268bb4e8a642383bee4be4835a/RecipeManager-base/src/main/java/haveric/recipeManager/recipes/WorkbenchEvents.java#L523

RecipeManager cancels the CraftItemEvent:
https://github.com/haveric/RecipeManager2/blob/8b3884a27bee3c268bb4e8a642383bee4be4835a/RecipeManager-base/src/main/java/haveric/recipeManager/recipes/WorkbenchEvents.java#L533

RecipeManager subtracts ingredients from the crafting grid:
https://github.com/haveric/RecipeManager2/blob/8b3884a27bee3c268bb4e8a642383bee4be4835a/RecipeManager-base/src/main/java/haveric/recipeManager/recipes/WorkbenchEvents.java#L540
- `#subtractIngredients(Inventory, ItemResult, boolean)`:
   - https://github.com/haveric/RecipeManager2/blob/8b3884a27bee3c268bb4e8a642383bee4be4835a/RecipeManager-base/src/main/java/haveric/recipeManager/recipes/PreparableResultRecipe.java#L216
